### PR TITLE
Vector notebooks

### PR DESCRIPTION
**Pull request by rogerkuou**
Added the parts for vector data

### CODE REVIEW

**Comment by rogerkuou** [comment]
Hi @fnattino,  I added two semi-filled notebooks for vector data and the data I needed. I limited the sizes of data by pre-cropping.  Would you like to give a look?

File: notebooks/03-vector-data-in-python.ipynb [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    "path_wells = 'data/wells.zip'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load the two files useing the \"read_file\" function"
```
Comment (fnattino) [inline notebooks/03-vector-data-in-python.ipynb:48]:
```suggestion
    "# Load the two files using the \"read_file\" function"
```

File: notebooks/03-vector-data-in-python.ipynb [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    "### Visualize polygons and points "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Visualize fields using the \"plot\" function"
```
Comment (fnattino) [inline notebooks/03-vector-data-in-python.ipynb:74]:
```suggestion
    "# Visualize fields using the \"plot\" method"
```

File: notebooks/03-vector-data-in-python.ipynb [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+    "### Crop vector data by bounding box"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Align the CRS by \"to_crs\" function"
```
Comment (fnattino) [inline notebooks/03-vector-data-in-python.ipynb:100]:
```suggestion
    "# Align the CRS by \"to_crs\" method"
```

File: notebooks/03-vector-data-in-python.ipynb [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get the bounding box of the fields using the \"total_bounds\" attributes"
```
Comment (fnattino) [inline notebooks/03-vector-data-in-python.ipynb:109]:
```suggestion
    "# Get the bounding box of the fields using the \"total_bounds\" attribute"
```

File: notebooks/03-vector-data-in-python.ipynb [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Select the wells withing the bounding box using the \"cx\" indexer "
```
Comment (fnattino) [inline notebooks/03-vector-data-in-python.ipynb:118]:
```suggestion
    "# Select the wells within the bounding box using the \"cx\" indexer "
```

File: notebooks/03-vector-data-in-python.ipynb [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+    "### Spatial query using `clip`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Grow a 500m buffer around wells using the \"buffer\" function\n",
```
Comment (fnattino) [inline notebooks/03-vector-data-in-python.ipynb:148]:
```suggestion
    "# Grow a 500m buffer around wells using the \"buffer\" method\n",
```

File: notebooks/03-vector-data-in-python.ipynb [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+    "### Spatial query using `sjoin`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Convert wells buffer into a GeoDataFrame\n",
```
Comment (fnattino) [inline notebooks/03-vector-data-in-python.ipynb:184]: I have recently realised the Series' method `to_frame` also does this

**Review by fnattino — COMMENTED** [review]
Hi @rogerkuou, simply great!

Some general comments:
* Should we move the `data` folder outside `notebooks`?
* The wells file is provided as a zip file. Is that because you want to show that geopandas also handles zipped archive files? But maybe you can leave the double extension (calling it `wells.gpkg.zip`) so that it is clear what the content of the archive is?
* I think I will show the data access part, but then upload some resized raster files to the repository, to avoid potentially long downloading time. Thus, will delete the search json file and will need a small update of notebook 4.

File: notebooks/03-vector-data-in-python.ipynb (hunk 7 above)
Comment (rogerkuou) [reply notebooks/03-vector-data-in-python.ipynb:184]: Thx for the tips! It just the `to_frame` function won't automatically give the column name `geometry` hence this...

**Comment by rogerkuou** [comment]
Thx @fnattino , I have applied most of your comments.
I removed the assumption on `search.json` in part 4. Will use what you have created before hand.

**Pull request merged**